File: ocaml/src/GRL_6_A.ml
(*+ import include/template/template.ml +*)
(*+ import include/graph/max_flow.ml +*)

let () =
  let v = read_int () in
  let e = read_int () in
  let g = A.make v [] in
  for i = 0 to e - 1 do
    let u = read_int () in
    let v = read_int () in
    let c = read_int () in
    add_edge g u v c 0
  done;
  let f = max_flow g 0 (v - 1) 0 max_int (+) (-) in
  pf "%d\n" f

File: cpp/src/GRL_6_A.cpp
#include "../include/graph/max_flow.cpp"
#include "../include/graph/capacity_graph.hpp"

using namespace std;

int main() {
  int V, E, u, v, c;
  scanf("%d%d", &V, &E);
  CapacityGraph<int> g(V);
  while (E--) {
    scanf("%d%d%d", &u, &v, &c);
    add_edge(g, u, v, c);
  }
  printf("%d\n", max_flow(g, 0, V - 1));
  return 0;
}
